lib/std/meta.zig

```diff
@@ -628,7 +628,7 @@ pub fn DeclEnum(comptime T: type) type {
     }
     return @Type(.{
         .@"enum" = .{
-            .tag_type = std.math.IntFittingRange(0, fieldInfos.len - 1),
+            .tag_type = std.math.IntFittingRange(0, if (fieldInfos.len == 0) 0 else fieldInfos.len - 1),
```

Comment:
what about using `-|`

Reply:
I'm afraid I'm not smart enough to use that kind operator. : )

Actually I have been working a lot with `std.meta` recently and noticed that in other analogs of this function (e.g. `std.meta.FieldEnum`) the "empty" case is handled by just returning an empty enum at the top of the function ([here](https://github.com/ziglang/zig/blob/master/lib/std/meta.zig#L527)) so maybe that would have been better here for consistency but oh well, too late now and I don't think the merged change is ambiguous anyway.